### ISSUE TITLE
Set user problem

### ISSUE THREAD

**Issue by microdesign** [tymondesigns/jwt-auth#968]
`        $user = User::find($id);

        JWTAuth::fromUser($user);
        dd(JWTAuth::getToken());`

the result is 'false' 
user object is there I check it already.

**Comment by knox2** [comment]
How have you checked it

**Comment by microdesign** [comment]
I mean I check the user object with dd($user); in case I send empty object to fromUser method.

But the user var contain the data from the model and the correct user is loaded.
But the method doesnt seems to doing something.

**Comment by knox2** [comment]
can you paste the result of dd($user)

**Comment by microdesign** [comment]
I already did it 

"the result is 'false'"

**Comment by knox2** [comment]
You've said dd($user) returns an object?

**Comment by microdesign** [comment]
yes, return a User model object from Laravel

**Comment by knox2** [comment]
which version of jwt

**Comment by microdesign** [comment]
"tymon/jwt-auth": "^0.5.6",

**Comment by knox2** [comment]
Then it should work, can you paste your controller?

**Comment by microdesign** [comment]
OK 

`class AdminUserController extends AdminGeneralController
{

     public function logginAs($id)
    {

        \Session::set('loggin_as_admin', $id);

        $user = User::find($id);

        JWTAuth::fromUser($user);
        dd(JWTAuth::getToken());
        
        return redirect('/#/dashboard');
    }
}
`

**Comment by knox2** [comment]
Its false because at that point you arent passing in the token via the request yet..

Do this 

 ```
public function logginAs($id)
{

    \Session::set('loggin_as_admin', $id);

    $user = User::find($id);

    $token  = JWTAuth::fromUser($user);
    dd($token);
    
    return redirect('/#/dashboard')->with('token'=>$token);
}
```

**Comment by microdesign** [comment]
Yes generate token now, but in the API where we redirect responses are 

{"error":"token_not_provided"}

maybe this is the problem because I redirect to API

**Comment by knox2** [comment]
In the API attach the token eg API?token=token_provided_from_user or add an Authorization header with value 'Bearer token_provided_from_user'

**Comment by microdesign** [comment]
Thanks I`ll try it

**Comment by microdesign** [comment]
I solve it by another way - thanks a lot for help

**Comment by ChromaticMind** [comment]
hey @microdesign, 

How did you solve this? 

I have the exact same issue. after `$token  = JWTAuth::fromUser($user);`  when I do `dd(JWTAuth::getToken());` it returns `false`.

This is part of an API call so not sure how to recall the same endpoint now with the token - don't think that is good practice anyway.

It would be great if you can share how you solved this issue.

Many thanks!

**Comment by microdesign** [comment]
I don't remember the problem already but I solve it by adding 

$user = User::find($user_id);
$token = JWTAuth::fromUser($user);

and your token is in the $token variable not use the method getToken()

**Comment by ChromaticMind** [comment]
Hey!

Yes that is precisely the problem @microdesign. I need `JWTAuth:: getToken();` to return the correct token. I am doing this

$user = User::find($id); 
*$token = JWTAuth::fromUser($user); 
JWTAuth::setToken($token);*
dd(JWTAuth::getToken());

setToken says "The token could not be parsed from the request" and that last part returns false. 

This is happening because the token is created by JWTAuth but it is not set in the header as the request has already come in. It seems like setToken called parseToken which instead of picking up the manually supplied token tries to pull it from $request and fails.

Any ideas?

**Comment by microdesign** [comment]
I can`t be sure my case is different from your - maybe the creator can answer you this

**Comment by williansmartins** [comment]
I have a similar issue, but the strange is when I run the server with "php artisan serve" it is ok, I can signin and logout, but when I put my code inside a Apache I can login but I get FALSE when I try logout

dd($JWTAuth->getToken()); //false

Anyone have a idea?

**Comment by williansmartins** [comment]
I discover the diference, I was send token like "Authorization Bearer {{token}}" this is not ok. When I send the token in URL= ".../api/v1/logout?token={{token}}" it works!. The strange is why is not possible send the token like anothers assignature? ("Authorization Bearer {{token}}")

**Comment by williansmartins** [comment]
I understood What happens, was about routes and midlewares configuration

**Comment by LarrySul** [comment]
here i have a problem which is when i try to access a protected route middle ware , I get token not provided , am doing a get request to get request which will then allow me fetch a specific user detail

	public function hostel()
	{
		{
		$user = JWTAuth::parseToken()->toUser();
		return response()->json( [
			'user' => $user
		], 200);
	}

here i have my route file

Route::get('/hostel', [
		'uses'=> 'AuthController@hostel',
		'middleware' => 'jwt.auth'
	]);

and here in my authentication controller I can't figure out how to fetch a authenticated user aside using this method

//here am trying to get a field of an authenticated user user jwt.auth
	public function getUser(Request $request)
	{
		$user = User::all();
		$response = [
			'user' => $user
		];
		return response()->json($response, 200);
	}


here i have my java script file
<script>
import axios from 'axios'
export default {
    data(){
        return {
            users: []
        }
    },
    created(){
        let token = localStorage.getItem('token');
        axios.get('/hostel?token =' + token)
        .then((response) =>{
          console.log(token);
        })
        .catch((error) => {
            console.log(error);
        }); 
    },
    mounted(){
        axios.get('/users')
        .then(response=>{
            this.users = response.data
        })
    },
     methods:{
        logout(){
        let token = localStorage.getItem('token');
        axios.get('/hostel?token=' + token)
        .then(response=>{
            localStorage.removeItem('token');
            this.$router.push('/home');
            });
        }
    }
}
</script>

**Comment by adarsh-bankopen** [comment]
I get this 

![image](https://user-images.githubusercontent.com/79446266/155703136-85270b59-a1c0-4949-916a-13f61e429a25.png)

when I run 

`php artisan serve
`
